src/sentry/workflow_engine/migrations/0051_migrate_remaining_issue_alerts.py

```diff
@@ -2241,16 +2231,12 @@ def migrate_issue_alert(rule: Any) -> None:
             )
             sentry_sdk.capture_exception(e)
 
-    migrated_issue_alerts = AlertRuleWorkflow.objects.filter(
-        rule_id=OuterRef("id"), rule_id__isnull=False
-    )
+    for rule in RangeQuerySetWrapperWithProgressBarApprox(Rule.objects.all()):
+        if AlertRuleWorkflow.objects.filter(rule_id=rule.id).exists():
```

Comment:
hm this is gonna be a lot queries 🤔 wonder if it makes sense to work through `Rule` in batches and pre-fetch the `AlertRuleWorkflow` for each batch to avoid timing out

Reply:
i can't pre-fetch because they're not connected by FK 😞